collect_app/src/main/java/org/odk/collect/android/upload/InstanceUploader.java

```diff
@@ -17,13 +17,30 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 
-import org.odk.collect.forms.instances.Instance;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.formmanagement.InstancesAppState;
+import org.odk.collect.android.injection.DaggerUtils;
 import org.odk.collect.android.utilities.InstancesRepositoryProvider;
+import org.odk.collect.forms.instances.Instance;
+import org.odk.collect.forms.instances.InstancesRepository;
 
 import java.util.ArrayList;
 import java.util.List;
 
+import javax.inject.Inject;
+
 public abstract class InstanceUploader {
+
+    @Inject
+    InstancesRepositoryProvider instancesRepositoryProvider;
+
+    @Inject
+    InstancesAppState instancesAppState;
+
+    public InstanceUploader() {
+        DaggerUtils.getComponent(Collect.getInstance()).inject(this);
```

Comment:
Look, grabbing the Collect singleton and using for injection in an object like this is bad, and I'm a bad person. We've done some much restructuring in the last few weeks though and starting to rework the submission/send class hierarchy doesn't feel like a great idea at the moment.

Reply:
![tumblr_24aa1561c944059f71984f454aa3af8b_12ca05de_250](https://user-images.githubusercontent.com/967540/115905591-282c5b80-a41b-11eb-830e-e544eea176b2.gif)